src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +265,225 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                        // From the relationship information, obtain the foreign key definition for the given target entity
+                        relationshipInfo.TargetEntityToFkDefinitionMap.TryGetValue(targetEntityName, out List<ForeignKeyDefinition>? listOfForeignKeys))
+                    {
+                        sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(sourceEntityName, out DatabaseObject? sourceDbo);
+                        // Find the foreignkeys in which the source entity is the referencing object.
+                        IEnumerable <ForeignKeyDefinition> sourceReferencingForeignKeysInfo =
+                            listOfForeignKeys.Where(fk =>
+                                fk.ReferencingColumns.Count > 0
+                                && fk.ReferencedColumns.Count > 0
+                                && fk.Pair.ReferencingDbTable.Equals(sourceDbo));
+
+                        sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(targetEntityName, out DatabaseObject? targetDbo);
+                        // Find the foreignkeys in which the target entity is the referencing object, i.e. source entity is the referenced object.
+                        IEnumerable<ForeignKeyDefinition> targetReferencingForeignKeysInfo =
+                            listOfForeignKeys.Where(fk =>
+                                fk.ReferencingColumns.Count > 0
+                                && fk.ReferencedColumns.Count > 0
+                                && fk.Pair.ReferencingDbTable.Equals(targetDbo));
+
+                        ForeignKeyDefinition? sourceReferencingFKInfo = sourceReferencingForeignKeysInfo.FirstOrDefault();
+                        if (sourceReferencingFKInfo is not null)
+                        {
+                            // When source entity is the referencing entity, FK directive is to be added to relationship fields
+                            // in the source entity.
+                            AppendFKDirectiveToReferencingFields(sourceFieldDefinitions, sourceReferencingFKInfo.ReferencingColumns);
+                        }
+
+                        ObjectTypeDefinitionNode targetObjectTypeDefinitionNode = objectTypes[targetEntityName];
+                        Dictionary<string, FieldDefinitionNode> targetFieldDefinitions = targetObjectTypeDefinitionNode.Fields.ToDictionary(field => field.Name.Value, field => field);
+                        ForeignKeyDefinition? targetReferencingFKInfo = targetReferencingForeignKeysInfo.FirstOrDefault();
+                        if (targetReferencingFKInfo is not null)
+                        {
+                            // When target entity is the referencing entity, FK directive is to be added to relationship fields
+                            // in the target entity.
+                            AppendFKDirectiveToReferencingFields(targetFieldDefinitions, targetReferencingFKInfo.ReferencingColumns);
```

Comment:
Do we need to check if the relationship is defined in the target entity too before adding the directive?

For example, take the example of Books-Publishers. Let's the relationship is defined only in the Publishers entity and not in the Books entity. 
So, in this case, the `publisher_id` column cannot be an optional field and will aways require a value. This is because, from Books entity, no relationship is defined to Publishers entity.

But, here, when the iteration is executing for Publishers entity, it will add a ForeignkeyDirective for `publisher_id` field in the Books entity which will eventually make it an optional field.

Reply:
We still want to allow a nested create on publisher which inserts a publisher and multiple books for that publisher, right? If we don't make publisher_id optional, how would we be able to allow such a mutation?

Reply:
Right. Thanks for the clarification.

This can be a test case for request validation - Neither `publisher_id` nor `publishers` is present in the request body. We need either one even though both are marked optional in the schema.

Reply:
yes